new-log-viewer/src/services/LogFileManager/utils.ts

```diff
@@ -11,60 +18,56 @@ import {getBasenameFromUrlOrDefault} from "../../utils/url";
 
 
 /**
- * Gets the log event number range [begin, end) of the page that starts at the given log event
- * index.
+ * Gets the data for the `PAGE_NUM` cursor.
  *
- * @param beginLogEventIdx
+ * @param pageNum
+ * @param eventPositionOnPage
  * @param numEvents
  * @param pageSize
- * @return An array:
- * - pageBeginLogEventNum
- * - pageEndLogEventNum
+ * @return Matching page and log event for the cursor (i.e. cursor data.)
  */
-const getPageBoundaries = (
-    beginLogEventIdx: number,
+const getPageNumCursorData = (
+    pageNum: number,
+    eventPositionOnPage: EVENT_POSITION_ON_PAGE,
     numEvents: number,
-    pageSize: number
-): [number, number] => {
-    const pageBeginLogEventNum: number = beginLogEventIdx + 1;
+    pageSize: number,
+): CursorData => {
+    const pageBeginIdx = (pageNum - 1) * pageSize;
+    const pageEndIdx = Math.min(numEvents, pageBeginIdx + pageSize);
 
-    // Clamp ending index using total number of events.
-    const pageEndLogEventNum: number = Math.min(numEvents + 1, pageBeginLogEventNum + pageSize);
+    const matchingIdx = eventPositionOnPage === EVENT_POSITION_ON_PAGE.TOP ?
+        pageBeginIdx :
+        pageEndIdx - 1;
 
-    return [
-        pageBeginLogEventNum,
-        pageEndLogEventNum,
-    ];
+    return {pageBeginIdx, pageEndIdx, matchingIdx};
 };
 
 /**
- * Gets the data for the `PAGE_NUM` cursor.
+ * Converts a potentially "invalid" `logEventIdx` into a valid log event index. `logEventIdx` may
+ * be "invalid" if:
+ * - `logEventIdx >= numEvents`.
+ * - `logEventIdx` excluded by the current filter.
```

Comment:
```suggestion
 * Gets the index, `i`, into the collection that contains the indices of the filtered log events
 * (let's call it `filteredLogEventIndices`) where:
 * - `i` is the largest index where `filteredLogEventIndices[i]` <= `logEventIdx`, or
 * - `i` is `0` if `logEventIdx < filteredLogEventIndices[0]`.
```

Reply:
True. Since we only call it in one place, can we simply inline this function? I feel like it's more confusing as a function since its return value is conceptually different depending on whether a filter is applied or not.